server/src/main/java/org/elasticsearch/rest/action/admin/indices/RestPutIndexTemplateAction.java

```diff
@@ -40,9 +44,14 @@ public String getName() {
 
     @Override
     public RestChannelConsumer prepareRequest(final RestRequest request, final NodeClient client) throws IOException {
-
         PutIndexTemplateRequest putRequest = new PutIndexTemplateRequest(request.param("name"));
-        putRequest.patterns(asList(request.paramAsStringArray("index_patterns", Strings.EMPTY_ARRAY)));
+        if (request.getRestApiVersion() == RestApiVersion.V_7 && request.hasParam("template")) {
+            deprecationLogger.compatibleApiWarning("template_parameter_deprecation",
+                "Deprecated parameter [template] used, replaced by [index_patterns]");
+            putRequest.patterns(List.of(request.param("template")));
```

Comment:
Can the message be a bit more descriptive ?
```
The deprecated field [template] was included in the request, API compatibility was used to map this field to [index_patterns]
```

Reply:
I could change it, sure, but I was aiming to have it be very much in line with the automatic logging we'll see from ParseField when api compatibility is invoked.

https://github.com/elastic/elasticsearch/blob/14aba7bcff63a630bc9a696c77fb45fa2acadfea/server/src/main/java/org/elasticsearch/common/xcontent/LoggingDeprecationHandler.java#L74-L81

So on the one hand we could be consistent with that (my original aim) or on the other we could be more descriptive. I don't have a strong preference -- which way would you prefer it?

Reply:
Let's leave this as-is and discuss in our next sync. No need to block this PR.